### PR TITLE
fix: Nav完善事件动作以及相关文档

### PR DESCRIPTION
**Pull request by wanglinfang2014**
### <samp>🤖 Generated by Copilot at 5c9f724</samp>

This pull request enhances the Menu component and its usage in the Nav renderer and the ConditionBuilderWithRemoteOptions value. It adds a new feature to handle the active menu items change with a callback function, fixes a bug of the submenu rendering, and allows passing custom class names to the menu items and submenus. It also refactors the Nav renderer and the ConditionBuilderWithRemoteOptions value to use the Menu component and adds the `data` prop to the event payloads.

### <samp>🤖 Generated by Copilot at 5c9f724</samp>

> _Oh we are the coders of the Menu component_
> _We add props and features with skill and intent_
> _We fix the bugs and we improve the style_
> _And we heave away on the `className` with a smile_

### <samp>🤖 Generated by Copilot at 5c9f724</samp>

*  Add the `onChange` prop to the Menu component and the Nav renderer, which allows passing a callback function that will be invoked when the active menu items change ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eR186-R187), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eR281-R283), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eL311-R327), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eL385-R402),  [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bR375-R380), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bL1192-R1206), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bR1255))
*  Add the `activeItems` variable to the Menu component, which holds an array of the active menu items and is passed to the `onChange` prop ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eL268-R270), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eL301-R315), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eR344), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eR368))
*  Modify the `onToggle` prop invocation in the Menu component to pass the link or the item object of the current menu item, which allows accessing the item data in the callback function ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eL496-R513))
*  Simplify the logic of rendering the expandIcon in the Menu component by using the Icon component with the icon prop, which avoids repeating the className and icon props for different cases ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eL519-R543))
*  Fix the issue of the submenu being rendered outside the screen in the Menu component by changing the rcMode value from 'vertical-right' to 'vertical-left' when the mode is 'float' and stacked is true ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eL640-R661))
*  Add the `className` prop to the MenuItem and SubMenu components, which allows passing a custom class name to the menu item and submenu components ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-c55e4e879a864d859ac0b4b876688e05bc08724d9e2ad118e8a185c83285f177L67-R68), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d73c859957e0e629801a44a04a35c3d27225812e79bcc778cb35634c87da3e79L212-R224))
*  Remove the `accordion` and `popupClassName` props from the NavItemSchema type and add them to the NavSchema interface, which are used to control the submenu accordion behavior and popup style in the Menu component ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bL86-R86), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bR263-R272))
*  Modify the comment of the `collapsed` prop in the NavSchema interface, which clarifies that it controls the menu collapse instead of the submenu collapse ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bL244-R240))
*  Remove the `getActiveItems` function and the logic of getting and comparing the active menu items from the Nav renderer, which are no longer needed as the active menu items are handled by the Menu component and passed to the Nav renderer via the `onChange` prop ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bL806-L824), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bL933-L940))
*  Add the `data` prop to the createObject function that creates the payload for the `loaded`, `collapsed`, and `toggled` events in the ConditionBuilderWithRemoteOptions value, which allows accessing the data variables in the event listener ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bL838-R839), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bL960-R954), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bL1040-R1035), [link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bL1064-R1069))
*  Add the `handleChange` method binding to the constructor of the ConditionBuilderWithRemoteOptions value, which ensures the correct `this` context when the method is invoked as a prop ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bR1002))
*  Remove the empty line in the Menu component, which is a minor code style improvement ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-51649a538e00907694676159197c60556bbe519882d3865eecb7c21103a0679eR527))
*  Remove the unused `dispatchEvent` and `store` props from the destructuring assignment of the `render` method of the ConditionBuilderWithRemoteOptions value, which are redundant ([link](https://github.com/baidu/amis/pull/7227/files?diff=unified&w=0#diff-d28c7ab328bb8a97e9bb5b69bf2344a707101eac02c186b7adc3a2fc061f582bL869-R865))

### CODE REVIEW

File: packages/amis/src/renderers/Nav.tsx [hunk 1]
```diff
@@ -835,12 +829,14 @@ const ConditionBuilderWithRemoteOptions = withRemoteConfig({
...
-
-    const rendererEvent = await dispatchEvent('loaded', {
-      data: response.value
-    });
+    const {dispatchEvent, data} = props;
 
+    const rendererEvent = await dispatchEvent(
+      'loaded',
+      createObject(data, {
+        items: response.links
```
Comment (hsm-lv) [inline packages/amis/src/renderers/Nav.tsx:837]: 直接干掉data的话，已经这么用的就会有问题，另外response里面的value和links有啥不同？

File: packages/amis/src/renderers/Nav.tsx [hunk 2]
```diff
@@ -957,12 +943,15 @@ const ConditionBuilderWithRemoteOptions = withRemoteConfig({
...
-    const rendererEvent = await dispatchEvent('loaded', {
-      data: ret.data,
-      item: {...item}
-    });
+    const {dispatchEvent, data} = props;
+
+    const rendererEvent = await dispatchEvent(
+      'loaded',
+      createObject(data, {
+        items: ret.data?.links,
```
Comment (hsm-lv) [inline packages/amis/src/renderers/Nav.tsx:951]: 同样

File: docs/zh-CN/components/nav.md [hunk 3]
```diff
@@ -323,38 +376,725 @@ order: 58
 }
 ```
 
-## 属性表
+## 悬浮导航
```
Comment (hsm-lv) [inline docs/zh-CN/components/nav.md:379]: 这几条没有说明用啥属性控制

File: docs/zh-CN/components/nav.md [hunk 4]
```diff
@@ -323,38 +376,725 @@ order: 58
...
+```
+
+## 动作表
+
+当前组件对外暴露以下特性动作，其他组件可以通过指定`actionType: 动作名称`、`componentId: 该组件id`来触发这些动作，动作配置可以通过`args: {动作配置项名称: xxx}`来配置具体的参数，详细请查看[事件动作](../../docs/concepts/event-action#触发其他组件的动作)。
+
+| 动作名称    | 动作配置                         | 说明           |
+| ----------- | -------------------------------- | -------------- |
+| updateItems | `value: string`或`value: item[]` | 更新导航数据源 |
+| reset       |                                  | 重置导航数据源 |
```
Comment (hsm-lv) [inline docs/zh-CN/components/nav.md:1011]: 这个reset是重置到什么数据呀？

**Review by hsm-lv — CHANGES_REQUESTED** [review]

File: packages/amis/src/renderers/Nav.tsx (hunk 1 above)
Comment (wanglinfang2014) [reply packages/amis/src/renderers/Nav.tsx:838]: 已修改，value是接口返回，控制选中的to值，不一定存在，links是菜单项数据源

File: packages/amis/src/renderers/Nav.tsx (hunk 2 above)
Comment (wanglinfang2014) [reply packages/amis/src/renderers/Nav.tsx]: 已修改

File: docs/zh-CN/components/nav.md (hunk 3 above)
Comment (wanglinfang2014) [reply docs/zh-CN/components/nav.md:381]: 已修改

File: docs/zh-CN/components/nav.md (hunk 4 above)
Comment (wanglinfang2014) [reply docs/zh-CN/components/nav.md:1021]: updateItems是更新数据源，reset是将更新后的数据源恢复为原始的，如果没有经过updateItems，执行reset是没有任何效果的

**Review by hsm-lv — APPROVED** [review]

**Review by lurunze1226 — APPROVED** [review]

**Pull request merged**